### PR TITLE
Add Sri Lanka

### LINKED ISSUE #2936: Data for Sri Lanka (LK) 🇱🇰
Now : "We do not yet have any information about this area." in https://www.electricitymap.org/zone/LK

Proposal : 
Add *daily* data from https://ceb.lk/ webpage or https://ceb.lk/electricity-generated/en webpage

[x] I have read the documentation about data sources https://github.com/tmrowco/electricitymap-contrib/wiki/Verify-data-sources

### PR DESCRIPTION
**Pull request by q--**
Closes #2936

Still needs bounding box generated, perhaps someone else could add it?

----

Capacity data sourced from the ["Long Term Generation Expansion Plan 2022-2041"](https://ceb.lk/front_img/img_reports/1636539187LTGEP_2022-2041_Web_compressed.pdf)

Renewable installed capacity taken from Figure 1.13 (private-owned) and Table 2.1 (government-owned hydro), except for biomass, where the higher figure mentioned in the section "5.4 Other Renewable Energy Development" was used. Together, these add up to within 1 MW of the stated total renewable capacity of 2,447 MW (which is mentioned in e.g. "1.4.2.1 Overview"). Solar includes rooftop solar; if desired, this could be changed to only utility-scale solar.

Thermal capactity taken from "2.2 Thermal Generation". Thermal Electricity generation is only from coal and oil, see Figure 1.14.
According to ["CEB - ANNUAL REPORT 2020" page 18 (20 in PDF)](https://ceb.lk/front_img/img_reports/164887150703-CEB-Annual_Report-2020-English.pdf), the 900MW Lakvijaya plant is the only coal plant in Sri Lanka, implying the others are oil-fired.
Installed capacity figures do not add up exactly to the 2,168 MW mentioned in the opening paragraph; I went with the combined total nameplate capacity from Table 2.5 (government-owned) and Table 2.6 (private-owned), adding up to 2197.9 MW. Then I put 900MW under coal and categorized the remainder as oil.

Pumped hydro is mentioned as being planned/investigated in several places, but currently isn't deployed. The same goes for battery storage.

Sri Lanka has no nuclear power plants. A report has been commissioned to investigate the possibility, see 4.2 (v).

---

*Marginally related:*

 I couldn't find the instructions for generating bounding boxes on the wiki, and the old README instructions I pulled from history I partially followed (I did try to install the npm dependencies; not going to install docker, not enough SSD space for something big I'll rarely use) didn't work. Perhaps the wiki can be updated, or if the instructions for bounding boxes are on the wiki a link added on the page with instructions for writing a new parser?

I also had trouble with testing the parser; the Python 3.7 that's supposedly supported didn't work because a dependency requires Python 3.8; perhaps [that wiki page](https://github.com/electricitymap/electricitymap-contrib/wiki/Set-up-local-environment) can be updated?

### CODE REVIEW

**Comment by q--** [comment]
> Sorry that this is a bit messy, we've been reworking how the geometries are created and that has left a bit of old, duplicate information about how to do it. 

To clarify, I couldn't find this information at all on the current wiki; instead, I did a `git checkout` of a file from a much older revision that had instructions.

Glad you're looking into fixing these.


----
If the bounding boxes are added, this PR is ready to merge.

File: parsers/LK.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import json
+from parsers.lib.exceptions import ParserException
+
+TIMEZONE_NAME = 'Asia/Colombo'
+GENERATION_BREAKDOWN_URL = "https://cebcare.ceb.lk/GenSum/GetLoadCurveData"
+SOURCE_NAME = "ceb.lk"
+
+def fetch_production(zone_key='LK', session=None, target_datetime=None, logger: logging.Logger = logging.getLogger(__name__)):
+  """Requests the last known production mix (in MW) of a given region
+  Arguments:
```
Comment (madsnedergaard) [inline parsers/LK.py:22]: Let's get rid of this comment from the example :)

**Review by madsnedergaard — COMMENTED** [review]
One very minor comment, otherwise LGTM!
The request is currently timing out when I'm testing here, don't know if source is just temporarily affected - did it work fine yesterday?

Screenshot 2022-04-05 at 09 48 33

**Comment by q--** [comment]
> One very minor comment, otherwise LGTM! The request is currently timing out when I'm testing here, don't know if source is just temporarily affected - did it work fine yesterday?

I'd hoped it'd be up again by now. Yes, it worked fine when I tested the parser last Saturday, but the entire subdomain has been down since at least yesterday.

**Comment by q--** [comment]
Still timing out. From what I've read, Sri Lanka is going through some major crisis right now, and one of the consequences is load shedding during much of the day and the electricity company being unable to afford fuel. I can see how that could lead to the customer portal server, which also hosts the power cut schedule, becoming overloaded.


Well, still hoping it comes back online. If you want to test and merge this before it comes back online so I don't have to check every day, you can change the URL to [this Archive.org](https://web.archive.org/web/20220124071533if_/https://cebcare.ceb.lk/GenSum/GetLoadCurveData) copy of generation details when testing.

**Comment by madsnedergaard** [comment]
Ah okay, thanks for the update - I tested and it looks fine with the archived url.
If you can remove the example comment then I'll merge it! :)

File: parsers/LK.py (hunk 1 above)
Comment (q--) [reply parsers/LK.py]: Done

**Review by madsnedergaard — APPROVED** [review]
Great! 🙌

**Pull request merged**